src/java.base/share/classes/java/lang/invoke/MethodHandleImpl.java

```diff
@@ -1275,8 +1274,8 @@ String internalProperties() {
         public MethodHandle asCollector(Class<?> arrayType, int arrayLength) {
             if (intrinsicName == Intrinsic.IDENTITY) {
                 MethodType resultType = type().asCollectorType(arrayType, type().parameterCount() - 1, arrayLength);
-                MethodHandle newArray = MethodHandleImpl.varargsArray(arrayType, arrayLength);
-                return newArray.asType(resultType);
+                MethodHandle collector = MethodHandleImpl.makeCollector(arrayType, arrayLength);
```

Comment:
Should `MethodHandleImpl.varargsArray` still be used here since it performs arity checking and caching? 

Maybe the arity checks are performed by `asCollectorType`, but that would still leave the caching.

Reply:
Ah right, good catch! This is a leftover from moving things around and should indeed use `varargsArray`.